migrations/20240712081151-current-collective-views-update.js

```diff
@@ -0,0 +1,278 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.query(`DROP VIEW IF EXISTS "CurrentCollectiveTransactionStats"`);
+
+    await queryInterface.sequelize.query(`DROP MATERIALIZED VIEW IF EXISTS "CollectiveTransactionStats"`);
+
+    await queryInterface.sequelize.query(`
+      CREATE MATERIALIZED VIEW "CollectiveTransactionStats" AS
+
+        WITH "ActiveCollectives" AS (
+          SELECT c."id" as "CollectiveId"
+          FROM "Transactions" t
+          LEFT JOIN "Collectives" c ON c."id" = t."CollectiveId" AND c."deletedAt" IS NULL
+          WHERE t."deletedAt" IS NULL
+            AND t."hostCurrency" IS NOT NULL
+            AND c."deactivatedAt" IS NULL
+            -- TODO: not sure why we have those conditions omn guest/incognito, too many accounts?
+            AND (c."data" ->> 'isGuest')::boolean IS NOT TRUE
+            AND c.name != 'incognito'
+            AND c.name != 'anonymous'
+            AND c."isIncognito" = FALSE
+          GROUP BY c."id"
+          HAVING COUNT(DISTINCT t."hostCurrency") = 1 -- no hostCurrency mismatch please!
+        )
+        SELECT
+          ac."CollectiveId" as "id",
+          MAX(t."createdAt") as "LatestTransactionCreatedAt",
+          COUNT(DISTINCT t.id) AS "count",
+
+          SUM(t."amountInHostCurrency")
+            FILTER (WHERE t.type = 'CREDIT' AND t."kind" NOT IN ('PAYMENT_PROCESSOR_COVER'))
+            AS "totalAmountReceivedInHostCurrency",
+
+          SUM(
+            COALESCE(t."amountInHostCurrency", 0) +
+            COALESCE(t."platformFeeInHostCurrency", 0) +
+            COALESCE(t."hostFeeInHostCurrency", 0) +
+            COALESCE(t."paymentProcessorFeeInHostCurrency", 0) +
+            COALESCE(t."taxAmount" * t."hostCurrencyFxRate", 0)
+          )
+            FILTER (
+              -- Get all credits related to orders
+              WHERE (t.type = 'CREDIT' AND t."OrderId" IS NOT NULL)
+              -- Deduct host fees and payment processor fees that are related to orders
+              OR (t.type = 'DEBIT' AND t.kind IN ('HOST_FEE', 'PAYMENT_PROCESSOR_FEE') AND t."OrderId" IS NOT NULL)
+            )
```

Comment:
I changed the definition here after `test/stories/ledger >Level 1: Same currency (USD) > 6. Expense marked as unpaid with Payment Processor fees NOT REFUNDED` surfaced another related issue. Taking this ledger as a reference:
![image](https://github.com/user-attachments/assets/8b531f61-c5ba-4eb7-be50-ccff2ecf251d)

The code was returning $1505 for the net total amount received. The reason was that it was summing the contribution ($1500) + the expense `PAYMENT_PROCESSOR_COVER` ($5). I chose to address that by filtering on `OrderId` not null, to stay simple and consistent. If we're worried about potential regressions, another option would be to be more specific:

```sql
WHERE (t.type = 'CREDIT' AND (t.kind !== 'PAYMENT_PROCESSOR_COVER' OR t."OrderId" IS NOT NULL))
```

@znarf Wdyt?

Reply:
> I chose to address that by filtering on OrderId not null, to stay simple and consistent.

This will exclude incoming expenses as money raised and I think this is gonna be a problem at some point in the future.